packages/harmony/src/components/button/FilterButton/FilterButton.tsx

```diff
@@ -4,190 +4,304 @@ import {
-    }, [variant, value, iconRight, onClick, onChange, onReset])
-
-    useEffect(() => {
-      if (isOpen) {
-        onOpen?.()
+  const filteredOptions = useMemo(
+    () =>
+      options?.filter(
+        ({ label }) =>
+          !filterInputValue ||
+          label?.toLowerCase().includes(filterInputValue.toLowerCase())
+      ),
+    [options, filterInputValue]
+  )
+
+  const optionElements = filteredOptions ? (
+    <FilterButtonKeyHandler
+      options={filteredOptions}
+      disabled={!isOpen}
+      onChange={handleChange}
+      optionRefs={optionRefs}
+      scrollRef={scrollRef}
+    >
+      {(activeValue) =>
+        filteredOptions.map((option, index) => (
+          <FilterButtonOption
+            ref={(el) => {
+              if (optionRefs && optionRefs.current && el) {
+                optionRefs.current[index] = el
+              }
+            }}
+            key={option.value}
+            option={option}
+            onChange={handleChange}
+            activeValue={value || activeValue}
+          />
+        ))
       }
-    }, [isOpen, onOpen])
-
-    const handleChange = useCallback(
-      (value: string) => {
-        setValue(value)
-        onChange?.(value)
-      },
-      [onChange, setValue]
-    )
-
-    const anchorRef = useRef<HTMLButtonElement>(null)
-
-    return (
-      <BaseButton
-        ref={ref || anchorRef}
-        styles={{
-          button: buttonCss,
-          icon: iconCss
-        }}
-        onClick={handleClick}
-        iconRight={Icon}
-        disabled={disabled}
-        aria-haspopup='listbox'
-        aria-expanded={isOpen}
+    </FilterButtonKeyHandler>
+  ) : null
+
+  return (
+    <BaseButton
+      ref={mergeRefs([ref, anchorRef])}
+      styles={{ button: buttonCss, icon: iconCss }}
+      onClick={handleClick}
+      iconRight={Icon}
+      disabled={disabled}
+      aria-haspopup='listbox'
+      aria-expanded={isOpen}
+    >
+      {leadingElement}
+      {selectedLabel ? renderLabel(selectedLabel) : label}
+      <Popup
+        anchorRef={anchorRef}
+        ref={scrollRef}
+        isVisible={isOpen}
+        onClose={() => setIsOpen(false)}
+        {...popupProps}
+        css={[{ overflowY: 'auto' }, popupProps?.css]}
       >
-        {leadingElement}
-        {label}
-        {children?.({
-          isOpen,
-          setIsOpen,
-          handleChange,
-          anchorRef
-        })}
-      </BaseButton>
-    )
-  }
-)
+        <Paper mt='s' border='strong' shadow='far'>
+          {children ? (
+            children({ onChange: handleChange, options: optionElements })
+          ) : (
+            <Flex
+              direction='column'
+              p='s'
+              gap='s'
+              alignItems='flex-start'
+              role='listbox'
+              aria-label={selectedLabel ?? label ?? props['aria-label']}
+              aria-activedescendant={selectedLabel}
+            >
+              {showFilterInput && filterInputProps ? (
```

Comment:
`filterInputProps` is required to show the filter input?

Reply:
since label is required, what you think?

Reply:
Gotcha, maybe just a comment on the type to call that out. We could even write the type so it requires `filterInputProps` when `showFilterInput` is true, but that may be overkill